tests/memoize-unique.ts

```diff
@@ -0,0 +1,92 @@
+/**
+ * based on https://github.com/faker-js/faker/issues/1785#issuecomment-1407773744
+ * replace this code with memoize-unique when it is published
+ */
```

Comment:
@kentcdodds 

faker.helpers.unique will be remove in the `faker` next version.
this is the replacement proposed by one of `faker` maintainer